File: packages/server-core/src/lib/movies/entities/movie-role-entity.ts
import { EntityBase } from '../../common/entities/entity-base';
import { Column, Entity, ManyToOne } from 'typeorm';
import { MovieEntity } from './movie-entity';
import { PersonEntity } from './person-entity';
import { MovieRole } from './movie-role';

@Entity()
export class MovieRoleEntity extends EntityBase {
  @Column('enum')
  role: MovieRole;

  @ManyToOne(() => MovieEntity, (movie) => movie.roles)
  movie: MovieEntity;

  @ManyToOne(() => PersonEntity, (person) => person.roles)
  person: PersonEntity;

  static create(
    role: MovieRole,
    person: PersonEntity,
    movie: MovieEntity
  ): MovieRoleEntity {
    const movieRole = new MovieRoleEntity();
    movieRole.role = role;
    movieRole.movie = movie;
    movieRole.person = person;
    movie.addRole(movieRole);
    person.addRole(movieRole);
    return movieRole;
  }
}

File: src/House.Flix.Core/Movies/Entities/MovieRoleEntity.cs
using House.Flix.Core.Common.Entities;
using House.Flix.Core.People.Entities;

namespace House.Flix.Core.Movies.Entities;

public class MovieRoleEntity : IEntity
{
    public Guid Id { get; set; }

    public PersonEntity Person { get; set; } = null!;
    public RoleEntity Role { get; set; } = null!;
    public MovieEntity Movie { get; set; } = null!;
}
